### PR TITLE
t3c parent.config fix for MSO non-topo mid cache group no parent cache groups

### PR DESCRIPTION
**Pull request by traeak**
Bug fix for #7137 (create topology on the fly for non topo delivery services)
Case:
delivery service with MSO but no topology.
mid cache group does not have primary and secondary parents assigned.
origins belong to any number of origin cache groups.
In this case the origin from only the fist encountered cache group was getting pulled into parent.config

<hr/>

## Which Traffic Control components are affected by this PR?

- Traffic Control Cache Config (`t3c`, formerly ORT)

## What is the best way to verify this PR?

Create delivery service, MSO non topo.  Ensure origins belong to different origin cache groups.  ensure the mid cache group has neither primary nor secondary cache groups assigned.  T3C run parent.config should 

## If this is a bugfix, which Traffic Control versions contained the bug?

- 7.1.0 (post #7137)

## PR submission checklist
- [x] This PR has tests 
- [ ] This PR has documentation -- BUG FIX!
- [x] This PR has a CHANGELOG.md entry 
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://apache.org/security) for details)

### CODE REVIEW

File: lib/go-atscfg/parentdotconfig.go [hunk 1]
```diff
@@ -167,8 +167,9 @@ func MakeParentDotConfig(
 
 // primarySecondary contains the names of the primary and secondary parent cache groups.
 type primarySecondary struct {
```
Comment (jpappa200) [inline lib/go-atscfg/parentdotconfig.go:169]: Should we update the comment for this structure to include what nulls is used for

File: lib/go-atscfg/remapdotconfig_test.go [hunk 2]
```diff
@@ -7759,7 +7759,6 @@ func TestMakeRemapDotConfigMidNoCacheRemapLineTopo(t *testing.T) {
 
 	dsType := tc.DSType("HTTP_NO_CACHE")
 
-	// BNO
```
Comment (jpappa200) [inline lib/go-atscfg/remapdotconfig_test.go:7762]: Random deletion, but looks like it needs to be.

**Review by jpappa200 — CHANGES_REQUESTED** [review]
LGTM, just a couple small things/questions but should be approved.

File: lib/go-atscfg/remapdotconfig_test.go (hunk 2 above)
Comment (traeak) [reply lib/go-atscfg/remapdotconfig_test.go:7762]: Unrelated cleanup while scanning for some debug things I throw in there.

**Review by jpappa200 — APPROVED** [review]
Looks good to me. 👍

**Review by zrhoffman — APPROVED** [review]
Looks good, tests pass! The CiaB failure is unrelated.

**Pull request merged**